src/amo/components/ScreenShots.js

```diff
@@ -1,41 +1,59 @@
-/* eslint-disable jsx-a11y/href-no-hash */
+/* global document, requestAnimationFrame, window
+ * eslint-disable jsx-a11y/href-no-hash */
 
 import React, { PropTypes } from 'react';
-
-import translate from 'core/i18n/translate';
+import { PhotoSwipeGallery } from 'react-photoswipe';
+import 'react-photoswipe/lib/photoswipe.css';
 
 import 'amo/css/ScreenShots.scss';
 
+const HEIGHT = 200;
+const WIDTH = 320;
 
-export class ScreenShotsBase extends React.Component {
+export default class ScreenShots extends React.Component {
   static propTypes = {
-    i18n: PropTypes.object.isRequired,
+    previews: PropTypes.array.isRequired,
   }
 
-  render() {
-    const { i18n } = this.props;
-    const totalImages = 5;
+  onClose = (photoswipe) => {
+    const index = photoswipe.getCurrentIndex();
+    const list = this.viewport.querySelector('.pswp-thumbnails');
+    const currentItem = list.children[index];
+    const offset = currentItem.getBoundingClientRect().x;
+    list.scrollLeft += offset - list.getBoundingClientRect().x;
+  }
 
-    // This is just a placeholder.
+  thumbnailContent = (item) => (
+    <img src={item.src} className="ScreenShots-image" height={HEIGHT} width={WIDTH} alt="" />
+  )
+
+  render() {
+    const { previews } = this.props;
+    const items = previews.map((preview) => ({
+      src: preview.image_url,
+      thumbnail_src: preview.thumbnail_url,
+      h: HEIGHT,
+      w: WIDTH,
+      title: preview.caption,
+    }));
+    const photoSwipeOptions = {
```

Comment:
Setting this in the `render()` when it doesn't ever change based on props seems odd to me; I don't _really_ think the performance difference matters, but it just doesn't seem like it needs to be here.

If it could be moved out that'd be cool.